modules/rank-eval/src/main/java/org/elasticsearch/index/rankeval/RankEvalRequest.java

```diff
@@ -64,16 +68,33 @@ public void setRankEvalSpec(RankEvalSpec task) {
         this.rankingEvaluationSpec = task;
     }
 
+    /**
+     * Sets the indices the search will be executed on.
+     */
+    public RankEvalRequest setIndices(String... indices) {
+        Objects.requireNonNull(indices, "indices must not be null");
+        for (String index : indices) {
+            Objects.requireNonNull(index, "index must not be null");
+        }
+        this.indices = indices;
+        return this;
+    }
+
+    public String[] getIndices() {
+        return Arrays.copyOf(indices, indices.length);
```

Comment:
I wanted to protect the indices from accidental modifications, but since there is a setter for them, they are mutable anyway, so I changed this like you suggested.